<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Средство обеспечения целостности</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabs">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabFiles">
       <attribute name="title">
        <string>Файлы</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <widget class="QCheckBox" name="filesFilter">
          <property name="text">
           <string>Только файлы с нарушенной целостностью</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="tableFiles">
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>133</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Файл</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Алгоритм</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Контрольная сумма</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Дата добавления</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Последнее нарушение</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Действия</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="filesControls">
          <item>
           <widget class="QPushButton" name="filesCommitButton">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Применить</string>
            </property>
            <property name="icon">
             <iconset resource="main.qrc">
              <normaloff>:/svgIcon/check-circle.svg</normaloff>:/svgIcon/check-circle.svg</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="filesRollbackButton">
            <property name="text">
             <string>Отменить</string>
            </property>
            <property name="icon">
             <iconset resource="main.qrc">
              <normaloff>:/svgIcon/slash.svg</normaloff>:/svgIcon/slash.svg</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="filesLeftSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="filesBackButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Назад</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="filesPageSize">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="filesPerPage">
            <property name="text">
             <string>/стр.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="filesForwardButton">
            <property name="text">
             <string>Вперёд</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="filesRightSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="filesInfo">
            <property name="text">
             <string>Страница 0 из 0. Всего записей: 0</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabTables">
       <attribute name="title">
        <string>Таблицы</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <widget class="QLabel" name="headerTables">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Текущая база данных: подключение отсутствует</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="tablesFilter">
          <property name="text">
           <string>Только таблицы с нарушением целостности</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="tableTables">
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>133</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Таблица</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Алгоритм</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Контрольная сумма</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Дата добавления</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Целостность</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Действия</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="tablesControls">
          <item>
           <widget class="QPushButton" name="tablesCommitButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Применить</string>
            </property>
            <property name="icon">
             <iconset resource="main.qrc">
              <normaloff>:/svgIcon/check-circle.svg</normaloff>:/svgIcon/check-circle.svg</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="tablesRollbackButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Отменить</string>
            </property>
            <property name="icon">
             <iconset resource="main.qrc">
              <normaloff>:/svgIcon/slash.svg</normaloff>:/svgIcon/slash.svg</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="tablesLeftSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="tablesBackButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Назад</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="tablesPageSize">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="tablesPerPage">
            <property name="text">
             <string>/стр.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="tablesForwardButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Вперёд</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="tablesRightSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="tablesInfo">
            <property name="text">
             <string>Страница 0 из 0. Всего записей: 0</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabScript">
       <attribute name="title">
        <string>Скрипт</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="scriptEditWrapper">
            <property name="title">
             <string>Скрипт</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="spacing">
              <number>2</number>
             </property>
             <property name="leftMargin">
              <number>2</number>
             </property>
             <property name="topMargin">
              <number>2</number>
             </property>
             <property name="rightMargin">
              <number>2</number>
             </property>
             <property name="bottomMargin">
              <number>2</number>
             </property>
             <item>
              <widget class="QPlainTextEdit" name="scriptEdit">
               <property name="font">
                <font>
                 <family>Consolas</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="scriptResultWrapper">
            <property name="title">
             <string>Результаты</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <property name="spacing">
              <number>2</number>
             </property>
             <property name="leftMargin">
              <number>2</number>
             </property>
             <property name="topMargin">
              <number>2</number>
             </property>
             <property name="rightMargin">
              <number>2</number>
             </property>
             <property name="bottomMargin">
              <number>2</number>
             </property>
             <item>
              <widget class="QPlainTextEdit" name="scriptResult">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="font">
                <font>
                 <family>Consolas</family>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="file">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="openScript"/>
    <addaction name="saveScript"/>
    <addaction name="separator"/>
    <addaction name="close"/>
   </widget>
   <widget class="QMenu" name="actions">
    <property name="title">
     <string>Действия</string>
    </property>
    <addaction name="connectToDB"/>
    <addaction name="separator"/>
    <addaction name="addObject"/>
    <addaction name="separator"/>
    <addaction name="execute"/>
    <addaction name="clearResults"/>
   </widget>
   <widget class="QMenu" name="help">
    <property name="title">
     <string>Справка</string>
    </property>
    <addaction name="algorithmList"/>
    <addaction name="commandList"/>
   </widget>
   <addaction name="file"/>
   <addaction name="actions"/>
   <addaction name="help"/>
  </widget>
  <widget class="QToolBar" name="toolbar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="connectToDB"/>
   <addaction name="separator"/>
   <addaction name="addObject"/>
   <addaction name="separator"/>
   <addaction name="openScript"/>
   <addaction name="saveScript"/>
   <addaction name="separator"/>
   <addaction name="execute"/>
   <addaction name="clearResults"/>
  </widget>
  <action name="openScript">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="main.qrc">
     <normaloff>:/svgIcon/folder.svg</normaloff>:/svgIcon/folder.svg</iconset>
   </property>
   <property name="text">
    <string>Открыть скрипт</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="saveScript">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="main.qrc">
     <normaloff>:/svgIcon/save.svg</normaloff>:/svgIcon/save.svg</iconset>
   </property>
   <property name="text">
    <string>Сохранить скрипт</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="connectToDB">
   <property name="icon">
    <iconset resource="main.qrc">
     <normaloff>:/svgIcon/database.svg</normaloff>:/svgIcon/database.svg</iconset>
   </property>
   <property name="text">
    <string>Соединиться с базой данных</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="execute">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="main.qrc">
     <normaloff>:/svgIcon/play.svg</normaloff>:/svgIcon/play.svg</iconset>
   </property>
   <property name="text">
    <string>Выполнить скрипт</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="algorithmList">
   <property name="text">
    <string>Список алгоритмов</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="commandList">
   <property name="text">
    <string>Список команд ПОЯ</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="clearResults">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="main.qrc">
     <normaloff>:/svgIcon/delete.svg</normaloff>:/svgIcon/delete.svg</iconset>
   </property>
   <property name="text">
    <string>Очистить поле результатов</string>
   </property>
   <property name="toolTip">
    <string>Очистить поле результатов</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="addObject">
   <property name="icon">
    <iconset resource="main.qrc">
     <normaloff>:/svgIcon/plus-circle.svg</normaloff>:/svgIcon/plus-circle.svg</iconset>
   </property>
   <property name="text">
    <string>Добавить объект</string>
   </property>
   <property name="toolTip">
    <string>Добавить файл или таблицу</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="close">
   <property name="text">
    <string>Закрыть</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="main.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>filesBackButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>318</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabs</sender>
   <signal>currentChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeTab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>connectToDB</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>execute</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearResults</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filesForwardButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tablesBackButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>288</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tablesForwardButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filesPageSize</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePageSize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>401</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tablesPageSize</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>changePageSize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>401</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filesRollbackButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleRollback()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tablesRollbackButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleRollback()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filesCommitButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleCommit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tablesCommitButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleCommit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addObject</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>close</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveScript</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openScript</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>handleAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filesFilter</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterTable(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>398</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changePage()</slot>
  <slot>changeTab(int)</slot>
  <slot>handleAction()</slot>
  <slot>changePageSize()</slot>
  <slot>handleTableClick()</slot>
  <slot>handleRollback()</slot>
  <slot>handleCommit()</slot>
  <slot>updateScriptResult(QString)</slot>
  <slot>filterTable(int)</slot>
 </slots>
</ui>
